### PR TITLE
Allow deep equality check on EntitySet

### PR DESCRIPTION
**Pull request by tamargrey**
Passes the `deep` parameter for `EntitySet.__eq__` down to the Woodwork dataframe. 

#1424

### CODE REVIEW

File: featuretools/entityset/entityset.py [hunk 1]
```diff
@@ -20,7 +20,7 @@
 # import pandas.api.types as pdtypes
 
 
-# from featuretools.utils.wrangle import _check_timedelta
+from featuretools.utils.wrangle import _dataframes_equal
```
Comment (thehomebrewnerd) [inline featuretools/entityset/entityset.py:23]: Do we need this import now?

File: featuretools/tests/entityset_tests/test_es.py [hunk 2]
```diff
@@ -1777,6 +1777,60 @@ def test_entityset_time_type_equality(es):
...
+                           dataframe=es['sessions'].copy(),
+                           index='id',
+                           logical_types=es['sessions'].ww.logical_types,
+                           semantic_tags=get_df_tags(es['sessions']))
+    second_es.add_dataframe(dataframe_name='customers',
+                            dataframe=es['customers'].copy(),
+                            index='id',
+                            time_index='signup_date',
+                            logical_types=es['customers'].ww.logical_types,
+                            semantic_tags=get_df_tags(es['customers']))
```
Comment (thehomebrewnerd) [inline featuretools/tests/entityset_tests/test_es.py:1807]: Super minor nitpick, but could we reorder these so we either add the dataframes to the entitysets in the same order or build the first entityset and then the second. I spent some time trying to figure out if there is a reason that we add customers to the first es and then sessions to the second, and then realized it doesn't matter. I found it a little harder to initially follow that we are building equal entitysets with this order.

File: featuretools/demo/flight.py [hunk 3]
```diff
@@ -95,7 +95,7 @@ def make_es(data):
                         'taxi_in', 'taxi_out', 'air_time', 'dep_time']
 
     logical_types = {'flight_num': ww.logical_types.Categorical,
-                     'distance_group': ww.logical_types.Ordinal(order=[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]),
+                     'distance_group': ww.logical_types.Ordinal(order=[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11]),
```
Comment (thehomebrewnerd) [inline featuretools/demo/flight.py:98]: Is this change needed for something in this PR?

**Review by thehomebrewnerd — COMMENTED** [review]
Looks good - just a couple minor comments.

File: featuretools/demo/flight.py (hunk 3 above)
Comment (tamargrey) [reply featuretools/demo/flight.py:98]: nope, just was trying to look at this demo entityset and realized it was missing an element from the order

File: featuretools/entityset/entityset.py (hunk 1 above)
Comment (tamargrey) [reply featuretools/entityset/entityset.py:23]: nope, good call. removing

File: featuretools/tests/entityset_tests/test_es.py (hunk 2 above)
Comment (tamargrey) [reply featuretools/tests/entityset_tests/test_es.py:1807]: yep, changing to building each entityset at a time.

Comment (tamargrey) [reply featuretools/tests/entityset_tests/test_es.py:1807]: oops, just pushed

**Review by thehomebrewnerd — APPROVED** [review]
Looks good!
